src/cryptonote_basic/hardfork.cpp

```diff
@@ -266,11 +266,9 @@ bool HardFork::reorganize_from_chain_height(uint64_t height)
 bool HardFork::rescan_from_block_height(uint64_t height)
 {
   CRITICAL_REGION_LOCAL(lock);
-  db.block_txn_start(true);
-  if (height >= db.height()) {
-    db.block_txn_stop();
+  db_wtxn_guard wtxn_guard(&db);
```

Comment:
The old code used a read-only txn, you're using a writer here. Is this intentional?

Reply:
No. I put a writer on purpose, but only because I thought this was a writing function and did not spot the txn start arg was readonly. I'll fix.